doc/rtd/topics/module_creation.rst

```diff
@@ -0,0 +1,94 @@
+.. _module_creation:
+
+Module Creation
+***************
+
+Much of cloud-init functionality is provided by :ref:`modules<modules>`.
+All modules follow a similar layout in order to provide consistent execution
+and documentation. Use the example provided here to create a new module.
+
+Example
+=======
+.. code-block:: python
+
+    # This file is part of cloud-init. See LICENSE file for license information.
+    """Example Module: Shows how to create a module"""
+
+    from logging import Logger
+
+    from cloudinit.cloud import Cloud
+    from cloudinit.config.schema import MetaSchema, get_meta_doc
+    from cloudinit.distros import ALL_DISTROS
+    from cloudinit.settings import PER_INSTANCE
+
+    MODULE_DESCRIPTION = """\
+    Description that will be used in module documentation.
+
+    This will likely take multiple lines.
+    """
+
+    meta: MetaSchema = {
+        "id": "cc_example",
+        "name": "Example Module",
+        "title": "Shows how to create a module",
+        "description": MODULE_DESCRIPTION,
+        "distros": [ALL_DISTROS],
+        "frequency": PER_INSTANCE,
+        "examples": [
+            "cc_example: example1",
```

Comment:
might want to express this as something a bit different from cc_example so it doesn't imply some sort of requirement to match the current module name.
```suggestion
            "some_new_config_key: allowed_value",
```